### PR TITLE
feat: port split_ifs tactic

### PR DESCRIPTION
**Pull request by dwrensha**
This is intended to be as close as possible of a translation of the mathlib3 version: https://github.com/leanprover-community/mathlib/blob/12a7da107a39ef936c5d4a8cb873018ad05b07a6/src/tactic/split_ifs.lean

### CODE REVIEW

File: Mathlib/Tactic/SplitIfs.lean [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+namespace Mathlib.Tactic
+
+open Lean Elab.Tactic Parser.Tactic Lean.Meta
+
+attribute [split_ifs_reduction] if_pos if_neg dif_pos dif_neg if_congr
+
+/-- Simulates the `all_goals` tactic combinator.
+-/
+private def evalAllGoals : TacticM Unit → TacticM Unit := fun tac => do
```
Comment (semorrison) [inline Mathlib/Tactic/SplitIfs.lean:22]: This is already in `Mathlib/Tactic/Core.lean`, at the end.

**Comment by semorrison** [comment]
Do you know about [`src/Lean/Meta/Tactic/SplitIf.lean`](https://github.com/leanprover/lean4/blob/master/src/Lean/Meta/Tactic/SplitIf.lean) in core? Perhaps it doesn't have a user-facing interface.

If we're meant to be using it here, lets use it. Otherwise let's leave a comment explaining why not. (I don't know which at this point.)

File: Mathlib/Tactic/SplitIfs.lean (hunk 1 above)
Comment (dwrensha) [reply Mathlib/Tactic/SplitIfs.lean]: Ah, very good. Updated.

**Comment by dwrensha** [comment]
I'd like to reuse the existing code if we can get away with it. However, doing so might mean that we can less faithfully match the behavior of mathlib3's `split_ifs`.
For example, as this PR is currently written, with a discharger that handles `True.intro`, we have
```lean
example (p : Prop) [Decidable p] : if if ¬p then p else True then p else ¬p :=
by split_ifs with h1 h2
   · assumption
   · assumption
```
where the subgoals are
```
Goals (2)
case inl
p : Prop
inst : Decidable p
h1 : p
⊢ p

case inr
p : Prop
inst : Decidable p
h1 : ¬p
⊢ ¬p
```
This closely matches how mathlib3 behaviors on them same example (when translated into Lean 3).

If we switch to using the existing `SplitIf.getSimpContext` and `SplitIf.discharge?`, then the subgoals become
```
Goals (3)
case inl.inl
p : Prop
inst : Decidable p
h1 : p
h2 : True
⊢ p

case inl.inr
p : Prop
inst : Decidable p
h1 : p
h2 : ¬True
⊢ ¬p

case inr
p : Prop
inst : Decidable p
h1 : ¬p
⊢ ¬p
```
Because `if True else x y` no longer simplifies.
Maybe we consider that a discrepancy that we can live with? I wouldn't be surprised if it never actually shows up in mathlib proofs.
Or maybe we could update the Lean 4 core `SplitIf` code to more closely match?
(One thing I still don't understand where is where in the mathlib3 code the `True.intro` is actually being handled. The [discharger](https://github.com/leanprover-community/mathlib/blob/12a7da107a39ef936c5d4a8cb873018ad05b07a6/src/tactic/split_ifs.lean#L38) looks to me like it only handles `assumption` and `not_not_intro`.)

**Comment by dwrensha** [comment]
Pushed a change that uses `SplitIf.getSimpContext` so that we don't need to add a new simp attribute, but keeps the custom discharger for now, to match the mathlib3 behavior. Further code reuse is possible if we decide we don't want to match the mathlib3 behavior so closely, or if we find a way to modify the `SplitIf` stuff to make it match.

**Comment by dwrensha** [comment]
I notice that the usual continuous integration checks are not running on this PR. Is that because it's from a branch of https://github.com/dwrensha/mathlib4, rather than https://github.com/leanprover-community/mathlib4? If so, may I be granted permission to create/modify branches on the main repo, and should I resubmit this PR in that way?

**Comment by digama0** [comment]
It sets the simp set to the `split_if_reduction` simp set, which contains `if_pos if_neg dif_pos dif_neg if_congr`. Simp knows how to discharge `True` even if you don't say anything, because the way simp works is to simplify the goal to `True` and then apply `trivial` at the end.

**Comment by dwrensha** [comment]
> Simp knows how to discharge True even if you don't say anything...

That seems to be true in Lean 3, but in Lean 4 when I use `SplitIf.discharge?`, the `True` does not get discharged, and instead gets split like any other if-the-else condition into a `True` and `¬True` branch. See `h2` below:
```
case inl.inl
p : Prop
inst : Decidable p
h1 : p
h2 : True
⊢ p

case inl.inr
p : Prop
inst : Decidable p
h1 : p
h2 : ¬True
⊢ ¬p
```

**Comment by dwrensha** [comment]
Closing in favor of #508, which uses a branch of the main repo, to get CI working.

**Pull request closed without merging**